src/commands/no_channel_command.ts

```diff
@@ -0,0 +1,49 @@
+import Command from './command';
+import Message from '../message';
+import { UserRole } from '../user';
+import Channel from '../channel';
+
+/** A command independant from the channel it's used on. */
+export default class NoChannelCommand extends Command {
+  /** Creates a new no-channel command.
+   *
+   * @param name - The name and label of the command.
+   * @param description - The description of the command.
+   * @param trigger - The regex to trigger the command.
+   * @param action - The action to execute with the command.
+   * @param role - The required role to execute the command.
+   */
+  constructor(
+    name: string,
+    description: string,
+    label: string,
+    trigger: RegExp,
+    action: (message: Message, match: RegExpMatchArray) => Promise<void>,
+    role?: UserRole,
+  ) {
+    super(
+      name,
+      description,
+      // The label is independant of the channel
+      () => label,
+      // The help string is the label and the description
+      (channel, prefix) => `\`${prefix}${label}\` - ${description}`,
+      // The trigger is independant of the channel
+      () => trigger,
+      action,
+      role,
+    );
+  }
+
+  public findCmdLabel(command: Command, channel: Channel): string {
+    if (this.name === command.name) {
+      return this.channelLabel(channel);
+    }
+
+    return undefined;
```

Comment:
This is a problem. Typescript doesn't complain because undefined can be a string as well but I doubt this should be the expected behavior. I can see that somewhere else this function is used to map commands which would mean that you get undefined values in the array and you use filter to throw the odd one out.

The minimum change required here is to change the return type from `string` to `string | undefined` so that at least it's controlled

Reply:
Alright, I fixed that and moved the function to `CommandGroup`